ui/src/app/pages/data/data-set/export-dataset-modal.tsx

```diff
@@ -138,53 +145,118 @@ export const ExportDatasetModal = ({
         'The request cannot be completed. Please try again or contact Support in the left hand navigation'
       );
     }
-  }
+  };
+
+  const isNotebooksLoading = existingNotebooks === undefined;
+  const isCodePreviewLoading = showCodePreview && !codePreview;
 
-  function loadHtmlStringIntoIFrame(html) {
+  // Any state that necessitates disabling all modal controls.
+  const modalLoading =
```

Comment:
this name isn't *quite* true - maybe "shouldDisable"?

Reply:
I like that. Thanks!